web/source/kmwbase.ts

```diff
@@ -28,6 +28,20 @@
    Copyright 2017-2019 SIL International
 ***/
 namespace com.keyman {
+
+  export interface OptionType {
+    root: string;
+    resources: string;
+    keyboards: string;
+    fonts: string;
+    attachType: string; // 'auto', 'manual', 'default'
```

Comment:
You should be able to define this as `attachType: 'auto'|'manual'|'default'` yeah? Or better yet, define `OptionAttachType` enumeration accordingly?

Reply:
Per https://help.keyman.com/developer/engine/web/15.0/reference/core/init#init_options

I ended up using `undefined | 'auto' | 'manual'`

Reply:
I'm probably doing undefined wrong. Should this be?
```suggestion
      attachType?: 'auto' | 'manual';
```